cdap-ui/app/cdap/components/AbstractWidget/MultiStageSelector/index.tsx

```diff
@@ -72,13 +72,36 @@ const MultiStageSelectorBase: React.FC<IMultiStageSelectorProps> = ({
       </div>
     );
   }
+  const validatedValue = value
+    .toString()
+    .split(',')
+    .filter((stage) => {
+      return inputStages.find((s) => s.id === stage);
+    })
+    .join(',');
+  if (validatedValue !== value) {
+    /**
+     * This will happen when the user configured the plugin (say joiner)
+     * with initial two stages and then later drop one of those stages.
+     *
+     * Now the previously selected input stages are no longer valid.
+     * The validatedValue will only have stages that are still input to joiner
+     * and will update the value of the plugin automatically.
+     *
+     * If not for this we will show only valid stages but the actual plugin property
+     * will have all the stages.
+     */
+    onChange(validatedValue);
```

Comment:
Is there some store where this should be validated? Seems odd to call back when rendering.

Reply:
@njbriggs The validation happens couple of lines above. The validation is nothing but checking if the input stages specified in the value still present as input.

Say,
- The user connects a File and a GCS source to joiner 
- Configures joiner and chooses the two sources in the multistageselector
- Closes the joiner and renames or removes one of the input stages, say File source

Upon opening the joiner the MultiStageSelector should not include the removed source as it is not longer valid. Previously the select widget used to remove it from rendering (as it is not present in the list). 

Now we are removing it from the list and updating the value of plugin property to have only valid input stages.